pkg/ccl/crosscluster/logical/dead_letter_queue_test.go

```diff
@@ -41,8 +42,93 @@ import (
 const (
 	defaultDbName = "defaultdb"
 	publicScName  = "public"
+	dbAName       = "a"
 )
 
+func setupDLQTestTables(
+	ctx context.Context,
+	t *testing.T,
+	sqlDB *sqlutils.SQLRunner,
+	kvDB *kv.DB,
+	srv serverutils.TestServerInterface,
+) (
+	tableNameToDesc map[string]catalog.TableDescriptor,
+	srcTableIDToName map[descpb.ID]dstTableMetadata,
+	expectedDLQTables []string,
+	ie isql.Executor,
+) {
+	s := srv.ApplicationLayer()
+	sd := sql.NewInternalSessionData(ctx, s.ClusterSettings(), "" /* opName */)
+	ie = s.InternalDB().(isql.DB).Executor(isql.WithSessionData(sd))
+
+	sqlDB.Exec(t, `CREATE TABLE foo (a INT)`)
+
+	sqlDB.Exec(t, `CREATE SCHEMA baz`)
+	sqlDB.Exec(t, `CREATE TABLE baz.foo (a INT)`)
+	sqlDB.Exec(t, `CREATE SCHEMA bar_`)
+	sqlDB.Exec(t, `CREATE TABLE bar_.foo (a INT)`)
+	sqlDB.Exec(t, `CREATE SCHEMA bar`)
+	sqlDB.Exec(t, `CREATE TABLE bar._foo (a INT)`)
+
+	sqlDB.Exec(t, `CREATE DATABASE a`)
+	sqlDB.Exec(t, `CREATE SCHEMA a.baz`)
+	sqlDB.Exec(t, `CREATE TABLE a.public.bar (a INT)`)
+	sqlDB.Exec(t, `CREATE TABLE a.baz.foo (a INT)`)
+
+	dstTableMeta := []dstTableMetadata{
+		{
+			database: defaultDbName,
+			schema:   publicScName,
+			table:    "foo",
+			tableID:  1,
+		},
+		{
+			database: defaultDbName,
+			schema:   "baz",
+			table:    "foo",
+			tableID:  1,
```

Comment:
i've just realized that it's a bit odd to hardcode the destination tableID to 1 after creating all the destination table and schema above. In prod, this would be a bad bug.

It seems more realistic and clear to grab the tableID from each table created and use that in `dstTableMetadata`. We can assume the srcID is some arbitrary int.